accounts/abi/bind/bind.go

```diff
@@ -224,12 +232,30 @@ func Bind(types []string, abis []string, bytecodes []string, fsigs []map[string]
 		_, ok := isLib[types[i]]
 		contracts[types[i]].Library = ok
 	}
-	// Generate the contract template data content and render it
-	data := &tmplData{
-		Package:   pkg,
-		Contracts: contracts,
-		Libraries: libs,
-		Structs:   structs,
+
+	var data interface{}
+	var templateSource string
+
+	// Generate the contract template data according to contract type (precompile/non)
+	if isPrecompile {
```

Comment:
Would it be possible to create a separate function for binding precompiles and create helper functions to share between them rather than add an argument to the existing `Bind` function? Changing the signature of `Bind` may result in any code that relies on the golang ABI running into issues.

Reply:
my first attempt to use it as an separate function but then I was under impression that it would make it much harder for code migration.